feat: add stop orders to order types spec

barnabee opened the pull request:
close #1666  . Add stop orders.

jgsbennett committed:
feat: Add new cases, mostly around eth changes during outages. A few extra cases for PUP.

jgsbennett committed:
chore: Linting fixes.

jgsbennett committed:
Merge branch 'master' of github.com:vegaprotocol/specs-internal into eth_changes_during_outages

jgsbennett committed:
Merge branch 'master' of github.com:vegaprotocol/specs-internal into eth_changes_during_outages

jgsbennett committed:
fix: More linting issues.

jgsbennett committed:
Merge remote-tracking branch 'origin/master' into eth_changes_during_outages

gordsport committed:
fix: spelling and AC code lint errors

gordsport committed:
Merge branch 'master' into eth_changes_during_outages

gordsport committed:
fix: spelling

gordsport committed:
Merge branch 'eth_changes_during_outages' of github.com:vegaprotocol/specs into eth_changes_during_outages

gordsport committed:
Merge branch 'master' into eth_changes_during_outages

jgsbennett committed:
Merge branch 'master' into eth_changes_during_outages

TomMcL committed:
feat: Clarifying fees paid (or not paid) by post-only orders (#1635)

jgsbennett committed:
Merge branch 'master' into eth_changes_during_outages

jgsbennett committed:
Merge pull request #1615 from vegaprotocol/eth_changes_during_outages

feat: Add new cases, mostly around eth changes during outages. A few …

radoslaw-szpiech committed:
chore: remove invalid and unimplemented use cases

radoslaw-szpiech committed:
Merge pull request #1636 from vegaprotocol/chore/remove-invalid-acs

chore: remove invalid and unimplemented use cases

witgaw committed:
refactor: floor slippage

witgaw committed:
Merge pull request #1638 from vegaprotocol/floor-slippage

refactor: floor slippage

campbellssource committed:
User interface: add Post only and Reduce only (#1640)

* draft for SORD

* add AC for showing rejected reason

* typo

* execution flags on orders table

* add codes to SORD

* Update user-interface/7002-SORD-submit_orders.md

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

* add link to docs

---------

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

davidsiska-vega committed:
Clarify rewards if multisig contains a key it should not (#1643)

* Clarify rewards if multisig contains a key it should not

* fix: MD lint error

---------

Co-authored-by: gordsport <gordon@vegaprotocol.io>

Jiajia-Cui committed:
feat: update FOK MO reduce-only

Jiajia-Cui committed:
Merge pull request #1647 from vegaprotocol/FOK_MO

feat: update FOK MO reduce-only

jgsbennett committed:
fix: Add some more cancel order ACs.

jgsbennett committed:
Update protocol/0033-OCAN-cancel_orders.md

Co-authored-by: Tom <tom@vegaprotocol.io>

jgsbennett committed:
Merge pull request #1652 from vegaprotocol/cancel_partial_active

fix: Add some more cancel order ACs.

jgsbennett committed:
feat: Add a test for checking performance scores when checkpoint rest… (#1637)

* feat: Add a test for checking performance scores when checkpoint restart happens in a new epoch.

* fix: Typo.

* chore: Fix indentation (tabs/spaces).

jgsbennett committed:
feat: Add a test for checking performance scores when checkpoint rest… (#1637)

* feat: Add a test for checking performance scores when checkpoint restart happens in a new epoch.

* fix: Typo.

* chore: Fix indentation (tabs/spaces).

Vegaklaus committed:
Update 0069-VCBS-validators_chosen_by_stake.md (#1653)

Deleted half an AC as discussed on slack (as the number for the performance score didn't make sense here)

mattrussell36 committed:
add spec for current collateral

mattrussell36 committed:
add spec for current functionality fo fills data grid

mattrussell36 committed:
Update user-interface/7001-COLL-collateral.md

Co-authored-by: Michael <michael@vega.xyz>

mattrussell36 committed:
Update user-interface/7001-COLL-collateral.md

Co-authored-by: Michael <michael@vega.xyz>

mattrussell36 committed:
Update user-interface/7001-COLL-collateral.md

Co-authored-by: Michael <michael@vega.xyz>

mattrussell36 committed:
Update user-interface/7001-COLL-collateral.md

Co-authored-by: Michael <michael@vega.xyz>

mattrussell36 committed:
clarify total margin available

mattrussell36 committed:
add asset file

mattrussell36 committed:
Add ACs for asset detail dialog and link up with collateral spec

mattrussell36 committed:
Update user-interface/7005-FILL-fills.md

Co-authored-by: Michael <michael@vega.xyz>

mattrussell36 committed:
Update user-interface/7005-FILL-fills.md

Co-authored-by: Michael <michael@vega.xyz>

mattrussell36 committed:
Update user-interface/7005-FILL-fills.md

Co-authored-by: Michael <michael@vega.xyz>

mattrussell36 committed:
Make notional value of feels a could

mattrussell36 committed:
Update user-interface/7005-FILL-fills.md

Co-authored-by: Michael <michael@vega.xyz>

cdummett committed:
feat: update for distributing remainder of liq fee

cdummett committed:
fix: correct AC

mattrussell36 committed:
add clarification on user role for trades during continuos vs auction

mattrussell36 committed:
add suggested sort order per account type

mattrussell36 committed:
add ui specs for finding closed markets

mattrussell36 committed:
fix number sequence

cdummett committed:
refactor: address feedback

witgaw committed:
Amend price monitoring ACs (#1660)

* refactor: update ACs

* chore: tell approbation to leave me alone

* chore: try again

* refactor: rephrase AC

* refactor: clarify

* refactor: fix spelling

ettec committed:
feat: update datanode spec with functionality to rollback node and functionality to fetch csv files containing network history (#1659)

* feat: update datanode spec with functionality to rollback node and functionality to fetch csv files containing network history

* fix: spell checker

---------

Co-authored-by: gordsport <gordon@vegaprotocol.io>
Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

EVODelavega committed:
spec: draft for time-based mempool pruning (#1645)

* spec: draft for time-based mempool pruning

Signed-off-by: Elias Van Ootegem <elias@vega.xyz>

* spec: spelling

Signed-off-by: Elias Van Ootegem <elias@vega.xyz>

* Spelling + clarify how block height is determined

Signed-off-by: Elias Van Ootegem <elias@vega.xyz>

* spec: add some additional ACs, remove mention of TxV4, clarify that transaction data includes block hash in addition to block height

Signed-off-by: Elias Van Ootegem <elias@vega.xyz>

* complete acceptance criteria

Signed-off-by: Elias Van Ootegem <elias@vega.xyz>

* fix typo

Signed-off-by: Elias Van Ootegem <elias@vega.xyz>

---------

Signed-off-by: Elias Van Ootegem <elias@vega.xyz>
Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

campbellssource committed:
re-word ac

gordsport committed:
chore: Add decision notes template (#1658)

* chore: Add decision notes template

Added a decision template based loosely on the [Any Decision Records](https://adr.github.io/madr/) template

Also updated some MD linting to allow sibling duplicated header titles and double spacing to make reading easier.

* fix: Spelling error

cdummett committed:
Merge pull request #1655 from vegaprotocol/feat/liq_fee_distribution

feat: spec for distributing remainder of liquidity fee

gordsport committed:
fix: 005 AC code link (#1671)

mattrussell36 committed:
fix: typo in ac number

davidsiska-vega committed:
fix: mempool pruning with PoW only

gordsport committed:
fix: linting errors

witgaw committed:
refactor: add liquidation price estimate

witgaw committed:
refactor: add ACs

witgaw committed:
refactor: another AC

witgaw committed:
Merge pull request #1675 from vegaprotocol/liquidation-price

Add liquidation price estimate non-protocol spec

witgaw committed:
refactor: cosmetic change

witgaw committed:
Merge pull request #1676 from vegaprotocol/correction

refactor: cosmetic change

gordsport committed:
Merge branch 'master' into remove-ttl

davidsiska-vega committed:
Merge pull request #1673 from vegaprotocol/remove-ttl

fix: mempool pruning with PoW only

vega-paul committed:
Update 0045-DSRC-data_sourcing.md (#1674)

* Update 0045-DSRC-data_sourcing.md

* Update 0045-DSRC-data_sourcing.md

* Wording of timestamp AC

---------

Co-authored-by: AnExsomnis <86832315+AnExsomnis@users.noreply.github.com>
Co-authored-by: Barney Mannerings <b.mannerings@gmail.com>

barnabee committed:
WIP: add stop orders to order types spec

Signed-off-by: Barney Mannerings <b.mannerings@gmail.com>

barnabee committed:
Update 0014-ORDT-order_types.md

Typo re-fix

daro-maj committed:
Merge branch 'master' into user-interface/EWAL-cant-destroy-ethwallet-session

daro-maj committed:
Merge pull request #1664 from vegaprotocol/user-interface/EWAL-cant-destroy-ethwallet-session

user interface: Can not destroy metamask session from dapp

Jiajia-Cui committed:
feat: update typo in 0061

Jiajia-Cui committed:
Merge pull request #1682 from vegaprotocol/typo-0061

feat: update typo in 0061

Jiajia-Cui committed:
feat: add explaination on ersatz validators

barnabee committed:
Add spam protection to perps spec

Signed-off-by: Barney Mannerings <b.mannerings@gmail.com>

barnabee committed:
Update trailing stops: only % distance

Signed-off-by: Barney Mannerings <b.mannerings@gmail.com>

Jiajia-Cui committed:
feat: correct typos

Jiajia-Cui committed:
feat: typo

Vegaklaus committed:
Merge pull request #1683 from vegaprotocol/ersatz-validator-explain

feat: add explaination on ersatz validators

daro-maj committed:
Update user-interface/7005-FILL-fills.md

daro-maj committed:
Merge branch 'master' into 7005-FILL-fills

daro-maj committed:
Merge pull request #1657 from vegaprotocol/7005-FILL-fills

Add UI spec for current functionality of fills

daro-maj committed:
Merge branch 'master' into 7001-COLL-collateral

witgaw commented on protocol/0014-ORDT-order_types.md:
```diff
@@ -47,8 +47,61 @@ Notes on scope of current version of this spec:
+
+* Stop order submissions must include a trigger direction.
+Direction may be _rises above_ or _falls below_.
+_Rises above_ stops trigger if the last traded price is higher than the trigger level, and _falls below_ stops trigger if the last traded price is lower than the trigger level.
```
Would it take a lot away from the feature if we made this conditional on mark price instead to save some computation? Can of course start with last traded price and modify it if does become an issue.

barnabee replied:
I think it isn't great to do that, would be quite surprising for users I think. Let's start with last traded price, stops shouldn't be a huge amount of computation to evaluate as you can maintain them in a sorted data structure.

witgaw commented on protocol/0014-ORDT-order_types.md:
```diff
@@ -47,8 +47,61 @@ Notes on scope of current version of this spec:
+
+    * Stop orders will only be accepted from keys with either a non-zero open position or at least one active order.
+
+    * A network parameter will control the maximum number of stop orders per party (suggested initial value: between 4 and 10).
```
Could we not just accept stop orders with volume up to the current riskiest long / riskiest short?

barnabee replied:
You could still have a lot of them, then. The thing this is trying to control is the number of stop orders rather than the total volume of your stops.

witgaw replied:
Ah sure, position decimal places...

witgaw commented on protocol/0014-ORDT-order_types.md:
```diff
@@ -47,8 +47,61 @@ Notes on scope of current version of this spec:
+
+    * A network parameter will control the maximum number of stop orders per party (suggested initial value: between 4 and 10).
+
+    * (TODO: confirm this makes sense) If the trader's position size gets to zero or changes direction all their stop orders are cancelled.
```
Probably not, one can always have some other passive limit orders they wish to stop-loss-manage in case they fill.

barnabee replied:
The problem is that if your stuff changes direction then the stops that are currently reduce only (which is a condition of stops in this spec) wont apply because they no longer reduce.

I guess we could allow them to sit there and only evaluate whether they are reduce only at the time they are triggered…

The other option will be to add "OTO" (one triggers the other) stops so that you can connect a stop to an active order and it won't become an "active" stop order unless/until that trades. But idk if it is worth it…

Maybe not having this condition and just checking the reduce only at trigger time saves it, what do you think @witgaw?

witgaw replied:
That'd be fine, as you say the reduce only bit will guard against stop order filling on the wrong side of the position. We should "garbage collect" all of them if at any point party has 0 open volume and no limit orders posted, but that's implementation detail.

barnabee replied:
I think we should explicitly state they are always removed if you have zero position and orders not leave it to implementation but will allow change of direction to guard against wrong way stops. 

I'll update the spec.

dalebennett1992 committed:
add browser wallet spec as part of approbation set up

dalebennett1992 committed:
chore: fix typo and add browser specs to app.json

dalebennett1992 committed:
chore: add browser wallet to categories, hopefully in the right place

dalebennett1992 committed:
Merge pull request #1687 from vegaprotocol/chore/add-vega-wallet-specs

chore: add vegawallet specs

mattrussell36 committed:
Merge branch 'master' into 7001-COLL-collateral

mattrussell36 committed:
Merge pull request #1656 from vegaprotocol/7001-COLL-collateral

Add UI spec for current functionality surrounding collateral

gordsport committed:
chore: Update 0033-OCAN-cancel_orders.md (#1686)

jeremyletang committed:
feat: update BRIE specs (#1654)

* feat: update BRIE specs

Signed-off-by: Jeremy Letang <me@jeremyletang.com>

* chore: update spelling and linting

* feat: update BRIE specs

Signed-off-by: Jeremy Letang <me@jeremyletang.com>

* chore: update spelling and linting

* fix: spelling and lint and rebase

* fix: spelling

---------

Signed-off-by: Jeremy Letang <me@jeremyletang.com>
Co-authored-by: gordsport <gordon@vegaprotocol.io>
Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

gordsport committed:
chore: Update cosmic elevator branch from master (#1688)

* refactor: let max oi drop during auciton

* fix: duplicate AC codes in 0069-VCBS (#1536) (#1537)

* Update `cosmicelevator` branch from `master` branch (#1550)

* fix: bring market.liquidity.targetstake.triggering.ratio in line with implementation

* fix: duplicate AC codes in 0069-VCBS (#1536)

* chore: trailing spaces

* fix: remove the unused netparam; not liking that

* chore: remove untestable AC from 0018 (#1530)

* chore: reword 0063-VALK-013

* Update 0035-LIQM-liquidity_monitoring.md (#1539)

* Update 0035-LIQM-liquidity_monitoring.md

* Update 0035-LIQM-liquidity_monitoring.md

* fix AC numbering on 0035-LIQM-liquidity_monitoring.md (#1541)

* Update 0011-NP-CLIE-client-interaction-tests.md

* feat: allow any key to request multisg signers update bundle

* feat: clarify spam status

* refactor: let max oi drop during auciton

* Wording tweaks for spam protection spec (#1548)

* grammar etc fixes to spam protection
* fix linting errors
* chore: remove brackets
* docs: another round of rewording
* add related topics section

Co-authored-by: David Siska <62546419+davidsiska-vega@users.noreply.github.com>
Co-authored-by: wwestgarth <william@vega.xyz>
Co-authored-by: wwestgarth <westgarth.w@gmail.com>
Co-authored-by: Paul Webb <86229849+vega-paul@users.noreply.github.com>
Co-authored-by: RomanLeca <roman.leca@yahoo.com>
Co-authored-by: Witold <witold@vega.xyz>
Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

* Update cosmicelevator branch from master to keep up-to-date (#1560)

* fix: bring market.liquidity.targetstake.triggering.ratio in line with implementation

* fix: duplicate AC codes in 0069-VCBS (#1536)

* chore: trailing spaces

* fix: remove the unused netparam; not liking that

* chore: remove untestable AC from 0018 (#1530)

* chore: reword 0063-VALK-013

* Update 0035-LIQM-liquidity_monitoring.md (#1539)

* Update 0035-LIQM-liquidity_monitoring.md

* Update 0035-LIQM-liquidity_monitoring.md

* fix AC numbering on 0035-LIQM-liquidity_monitoring.md (#1541)

* Update 0011-NP-CLIE-client-interaction-tests.md

* feat: allow any key to request multisg signers update bundle

* feat: clarify spam status

* refactor: let max oi drop during auciton

* Wording tweaks for spam protection spec (#1548)

* grammar etc fixes to spam protection
* fix linting errors
* chore: remove brackets
* docs: another round of rewording
* add related topics section

* fix spelling in spam protection spec (#1553)

* Update 0069-VCBS-validators_chosen_by_stake.md

Added pending validator to VCBS-59

* Update 0069-VCBS-validators_chosen_by_stake.md

Clarification on 0059

* Update 0062-SPAM-spam_protection.md

* Update 0062-SPAM-spam_protection.md

fixed link to defaults.go

* Add quality checks to the makefile (#1556)

* fix: broken links in the specs (#1557)

* fix: remove broken kraken link

---------

Co-authored-by: David Siska <62546419+davidsiska-vega@users.noreply.github.com>
Co-authored-by: wwestgarth <william@vega.xyz>
Co-authored-by: wwestgarth <westgarth.w@gmail.com>
Co-authored-by: Paul Webb <86229849+vega-paul@users.noreply.github.com>
Co-authored-by: RomanLeca <roman.leca@yahoo.com>
Co-authored-by: Witold <witold@vega.xyz>
Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>
Co-authored-by: Vegaklaus <62546834+Vegaklaus@users.noreply.github.com>

* chore: Update cosmicelevator from master branch (#1564)

* fix: bring market.liquidity.targetstake.triggering.ratio in line with implementation

* fix: duplicate AC codes in 0069-VCBS (#1536)

* chore: trailing spaces

* fix: remove the unused netparam; not liking that

* chore: remove untestable AC from 0018 (#1530)

* chore: reword 0063-VALK-013

* Update 0035-LIQM-liquidity_monitoring.md (#1539)

* Update 0035-LIQM-liquidity_monitoring.md

* Update 0035-LIQM-liquidity_monitoring.md

* fix AC numbering on 0035-LIQM-liquidity_monitoring.md (#1541)

* Update 0011-NP-CLIE-client-interaction-tests.md

* feat: allow any key to request multisg signers update bundle

* feat: clarify spam status

* refactor: let max oi drop during auciton

* Wording tweaks for spam protection spec (#1548)

* grammar etc fixes to spam protection
* fix linting errors
* chore: remove brackets
* docs: another round of rewording
* add related topics section

* fix spelling in spam protection spec (#1553)

* Update 0069-VCBS-validators_chosen_by_stake.md

Added pending validator to VCBS-59

* Update 0069-VCBS-validators_chosen_by_stake.md

Clarification on 0059

* Update 0062-SPAM-spam_protection.md

* Update 0062-SPAM-spam_protection.md

fixed link to defaults.go

* Add quality checks to the makefile (#1556)

* fix: broken links in the specs (#1557)

* fix: remove broken kraken link

* refactor: clarify 0042-LIQF-018

* feat: fix non-protocol linting and add makefile (#1563)

* feat: fix non-protocol linting and add makefile

* fix: remove binary file

* fix: remove binary

* fix: revert ui changes

* fix: remove package files

* Update glossaries/distributed-ledger-glossary.md

---------

Co-authored-by: David Siska <62546419+davidsiska-vega@users.noreply.github.com>
Co-authored-by: wwestgarth <william@vega.xyz>
Co-authored-by: wwestgarth <westgarth.w@gmail.com>
Co-authored-by: Paul Webb <86229849+vega-paul@users.noreply.github.com>
Co-authored-by: RomanLeca <roman.leca@yahoo.com>
Co-authored-by: Witold <witold@vega.xyz>
Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>
Co-authored-by: Vegaklaus <62546834+Vegaklaus@users.noreply.github.com>

* chore: Update cosmicelevator branch from master (#1573)

* docs: post only orders (#1562)

* docs: Adding specs for post-only orders

* Update from master (#1602)

* fix: bring market.liquidity.targetstake.triggering.ratio in line with implementation

* fix: duplicate AC codes in 0069-VCBS (#1536)

* chore: trailing spaces

* fix: remove the unused netparam; not liking that

* chore: remove untestable AC from 0018 (#1530)

* chore: reword 0063-VALK-013

* Update 0035-LIQM-liquidity_monitoring.md (#1539)

* Update 0035-LIQM-liquidity_monitoring.md

* Update 0035-LIQM-liquidity_monitoring.md

* fix AC numbering on 0035-LIQM-liquidity_monitoring.md (#1541)

* Update 0011-NP-CLIE-client-interaction-tests.md

* feat: allow any key to request multisg signers update bundle

* feat: clarify spam status

* refactor: let max oi drop during auciton

* Wording tweaks for spam protection spec (#1548)

* grammar etc fixes to spam protection
* fix linting errors
* chore: remove brackets
* docs: another round of rewording
* add related topics section

* fix spelling in spam protection spec (#1553)

* Update 0069-VCBS-validators_chosen_by_stake.md

Added pending validator to VCBS-59

* Update 0069-VCBS-validators_chosen_by_stake.md

Clarification on 0059

* Update 0062-SPAM-spam_protection.md

* Update 0062-SPAM-spam_protection.md

fixed link to defaults.go

* Add quality checks to the makefile (#1556)

* fix: broken links in the specs (#1557)

* fix: remove broken kraken link

* refactor: clarify 0042-LIQF-018

* feat: fix non-protocol linting and add makefile (#1563)

* feat: fix non-protocol linting and add makefile

* fix: remove binary file

* fix: remove binary

* fix: revert ui changes

* fix: remove package files

* Update glossaries/distributed-ledger-glossary.md

* Update 0069-VCBS-validators_chosen_by_stake.md

Added ACs to match the implementation so its codified (i.e., if an Ersatzvalidator drops below ownstake and there's no-one there to replace it, it is not demoted to pending.

* Update 0069-VCBS-validators_chosen_by_stake.md

* fix: Update acceptance criteria to correctly reflect implementation (#1549)

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

* fix: update epoch ACs (#1555)

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

* feat: update 0042, LP fee Transfer, Distribution

* Add build folder to gitigore

The approbation checks in `makefile` use this folder. Make sure it's not checked in.

* Update protocol/0042-LIQF-setting_fees_and_rewarding_lps.md

Co-authored-by: David Siska <62546419+davidsiska-vega@users.noreply.github.com>

* Ignore dictionary

Spellcheck creates a dictionary file. Ensure it isn't committed.

* Un parallelise clone-sources in makefile

The makefile tried to be clever and run all the remote-repo clones for Approbation checks in parallel. Due to my limited makefile skills, the check could run before a clone was finished. Run them in series instead.

This also improves the output a little

* Add specs param to names and codes in makefile

The old behaviour (default paths) should have been deprecated ages ago, but hasn't been. This change makes it provide the same paths as the defaults.

* feat: update maker fee transfer account

* Update 0069-VCBS-validators_chosen_by_stake.md

* Update protocol/0029-FEES-fees.md

Co-authored-by: David Siska <62546419+davidsiska-vega@users.noreply.github.com>

* Update protocol/0029-FEES-fees.md

Co-authored-by: David Siska <62546419+davidsiska-vega@users.noreply.github.com>

* Update protocol/0029-FEES-fees.md

Co-authored-by: David Siska <62546419+davidsiska-vega@users.noreply.github.com>

* Update 0048-DSRI-data_source_internal.md

* fix: refactor 0069-VCBS-052 (#1570)

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>
Co-authored-by: Vegaklaus <62546834+Vegaklaus@users.noreply.github.com>

* fix: MD and spelling lint checks

* fix: MD and spelling lint errors

* fix: update wording for AC 0059-VCBS-016- (#1566)

* Update 0062-SPAM-spam_protection.md (#1571)

* Update 0062-SPAM-spam_protection.md

* Update protocol/0062-SPAM-spam_protection.md

* Update protocol/0062-SPAM-spam_protection.md

---------

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

* Update 0011-NP-CLIE-client-interaction-tests.md (#1542)

* Update 0011-NP-CLIE-client-interaction-tests.md

Added ACs to test static spam protection (wallet doesn't allow you to get banned)

* Update non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

* Update non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

* Update non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

* Update non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

* Update non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

* Update non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

* Update 0011-NP-CLIE-client-interaction-tests.md

* Update non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

* Update non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

* Update non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

* Update non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

* Update 0011-NP-CLIE-client-interaction-tests.md

* Update 0011-NP-CLIE-client-interaction-tests.md

* chore: fix spelling and MD lint checks

---------

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>
Co-authored-by: gordsport <gordon@vegaprotocol.io>

* Update 0011-NP-CLIE-client-interaction-tests.md (#1579)

* refactor: clarify AC

* fix: adding bound on slippage part of margin

* fix: adding bound on slippage part of margin - ACs

* chore: lint

* chore: format

* chore: add ac codes

* chore: lint

* feat: make it nonlinear

* fix: remove obsolete ac

* fix: typo

* fix: typo from comments

Co-authored-by: Tom <tom@vegaprotocol.io>

* fix: code block in list formatting (#1584)

* chore: Remove untestable acceptance criteria

* feat: add rate limit spec

* Update 0057-TRAN-transfers.md (#1588)

* Update 0057-TRAN-transfers.md

* Update 0057-TRAN-transfers.md

* chore: add AC codes

* Update 0057-TRAN-transfers.md (#1590)

* Update 0069-VCBS-validators_chosen_by_stake.md

Added Acs for IssueSignature

* Update 0069-VCBS-validators_chosen_by_stake.md

* fix: Expand acceptance criteria to make them clear for 0011-NP-CLIE-client-interaction (#1585)

* fix: Expand acceptance criteria to make them clear

* chore: Fix formatting

* fix: Address review comments

---------

Co-authored-by: Paul Webb <86229849+vega-paul@users.noreply.github.com>

* feat: Add new PLUP AC for state-sync in protocol upgrade.

* feat: add closeout API

* chore: tidy up

* chore: remove blank line

* chore: by more specific about when a client should be banned

* feat: update formula line 140

* fix: Add extra cases for various things that might be affected by validators joining and leaving.

* fix: Review comments for new ACs.

* fix: Satisfy the linter.

* refactor: add slippage to orders

* refactor: update examples

* chore: fix linter issues

* refactor: fix typo

* refactor: formlise margin level treatment

* refactor: amend/add ACs

* Update 0042-LIQF-setting_fees_and_rewarding_lps.md (#1598)

* Update 0042-LIQF-setting_fees_and_rewarding_lps.md

* fix: linting errors

---------

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>
Co-authored-by: gordsport <gordon@vegaprotocol.io>

* fix: linting error

* fix: MD lint and spelling

* Update 0069-VCBS-validators_chosen_by_stake.md (#1604)

* Update 0069-VCBS-validators_chosen_by_stake.md

* Update 0069-VCBS-validators_chosen_by_stake.md

* Update 0069-VCBS-validators_chosen_by_stake.md

* Update 0069-VCBS-validators_chosen_by_stake.md

* Update 0069-VCBS-validators_chosen_by_stake.md

* Update 0069-VCBS-validators_chosen_by_stake.md

* Update 0069-VCBS-validators_chosen_by_stake.md

* Update 0069-VCBS-validators_chosen_by_stake.md

* Update 0069-VCBS-validators_chosen_by_stake.md

* fix: MD linting and spelling errors

---------

Co-authored-by: gordsport <gordon@vegaprotocol.io>

* Update 0069-VCBS-validators_chosen_by_stake.md (#1596)

* Update 0069-VCBS-validators_chosen_by_stake.md

* fix: spelling errors

---------

Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>
Co-authored-by: gordsport <gordon@vegaprotocol.io>

* fix: More validator join/leave test cases involving checkpoints. (#1601)

* fix: More validator join/leave test cases involving checkpoints.

* fix: spelling error

* fix: Update AC codes to not be duplicates.

---------

Co-authored-by: gordsport <gordon@vegaprotocol.io>
Co-authored-by: Gordsport <83510148+gordsport@users.noreply.github.com>

* feat: update make file and add approabtion to CI (#1605)

* feat: update make file and add approabtion to CI

* chore: add more checks and fix errors

* fix: checkout PR branch

* fix: get branch

* fix: approbation errors

---------

Co-authored-by: David Siska <62546419+davidsiska-vega@users.noreply.github.com>
Co-authored-by: wwestgarth <william@vega.xyz>
Co-authored-by: wwestgarth <westgarth.w@gmail.com>
Co-authored-by: Paul Webb <86229849+vega-paul@users.noreply.github.com>
Co-authored-by: RomanLeca <roman.leca@yahoo.com>
Co-authored-by: Witold <witold@vega.xyz>
Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>
Co-authored-by: Vegaklaus <62546834+Vegaklaus@users.noreply.github.com>
Co-authored-by: MuthuVega <91467655+MuthuVega@users.noreply.github.com>
Co-authored-by: AnExsomnis <86832315+AnExsomnis@users.noreply.github.com>
Co-authored-by: Jiajia-Cui <jiajia@vega.xyz>
Co-authored-by: Edd <edd@vega.xyz>
Co-authored-by: Jiajia-Cui <92106936+Jiajia-Cui@users.noreply.github.com>
Co-authored-by: claudiumilea <claudius.milea@gmail.com>
Co-authored-by: Tom <tom@vegaprotocol.io>
Co-authored-by: MuthuVega <muthu@vega.xyz>
Co-authored-by: Philip Scott <phil@vega.xyz>
Co-authored-by: Philip Scott <94362991+pscott31@users.noreply.github.com>
Co-authored-by: Jake Bennett <jake@vegaprotocol.io>

* fix: lint errors and duplicate paragraph (#1630)

* fix: lint errors and duplicate paragraph

* fix: MD lint error

* Chore: Update cosmicelevator from master (#1646)

* feat: Add new cases, mostly around eth changes during outages. A few extra cases for PUP.

* chore: Linting fixes.

* fix: More linting issues.

* fix: spelling and AC code lint errors

* fix: spelling

* feat: Clarifying fees paid (or not paid) by post-only orders (#1635)

* chore: remove invalid and unimplemented use cases

* refactor: floor slippage

* User interface: add Post only and Reduce only (#1640)

* draft for SORD

* add AC for showing rejected reason

* typo

* execution flags on orders table

* add codes to SORD

* Update user-interface/7002-SORD-submit_orders.md

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

* add link to docs

---------

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

* Clarify rewards if multisig contains a key it should not (#1643)

* Clarify rewards if multisig contains a key it should not

* fix: MD lint error

---------

Co-authored-by: gordsport <gordon@vegaprotocol.io>

---------

Co-authored-by: Jake Bennett <jake@vegaprotocol.io>
Co-authored-by: Tom <tom@vegaprotocol.io>
Co-authored-by: Rado <szpiechrados@gmail.com>
Co-authored-by: Witold <witold@vega.xyz>
Co-authored-by: Michael <michael@vega.xyz>
Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>
Co-authored-by: David Siska <62546419+davidsiska-vega@users.noreply.github.com>

---------

Co-authored-by: Witold <witold@vega.xyz>
Co-authored-by: David Siska <62546419+davidsiska-vega@users.noreply.github.com>
Co-authored-by: wwestgarth <william@vega.xyz>
Co-authored-by: wwestgarth <westgarth.w@gmail.com>
Co-authored-by: Paul Webb <86229849+vega-paul@users.noreply.github.com>
Co-authored-by: RomanLeca <roman.leca@yahoo.com>
Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>
Co-authored-by: Vegaklaus <62546834+Vegaklaus@users.noreply.github.com>
Co-authored-by: Tom <tom@vegaprotocol.io>
Co-authored-by: MuthuVega <91467655+MuthuVega@users.noreply.github.com>
Co-authored-by: AnExsomnis <86832315+AnExsomnis@users.noreply.github.com>
Co-authored-by: Jiajia-Cui <jiajia@vega.xyz>
Co-authored-by: Edd <edd@vega.xyz>
Co-authored-by: Jiajia-Cui <92106936+Jiajia-Cui@users.noreply.github.com>
Co-authored-by: claudiumilea <claudius.milea@gmail.com>
Co-authored-by: MuthuVega <muthu@vega.xyz>
Co-authored-by: Philip Scott <phil@vega.xyz>
Co-authored-by: Philip Scott <94362991+pscott31@users.noreply.github.com>
Co-authored-by: Jake Bennett <jake@vegaprotocol.io>
Co-authored-by: Rado <szpiechrados@gmail.com>
Co-authored-by: Michael <michael@vega.xyz>

dalebennett1992 committed:
fix: fix prefix for browser wallet

dalebennett1992 committed:
chore: undo formatting change

dalebennett1992 committed:
chore: fixed wrong file..

dalebennett1992 committed:
chore: now undo my formatting for the other file

dalebennett1992 committed:
Merge branch 'master' into fix/fix-browser-wallet-spec-prefix

dalebennett1992 committed:
Merge pull request #1689 from vegaprotocol/fix/fix-browser-wallet-spec-prefix

fix: fix prefix for browser wallet

dalebennett1992 committed:
add missing AC

dalebennett1992 committed:
Merge pull request #1690 from vegaprotocol/chore/add-more-ac-browser-wallet

Add some missing AC

dalebennett1992 committed:
add more ac

dalebennett1992 committed:
typoe

dalebennett1992 committed:
Merge pull request #1692 from vegaprotocol/chore/add-network-and-keypair-ac

Network and keypair AC for browser wallet

dalebennett1992 committed:
chore: add ac for the sprint

dalebennett1992 committed:
Merge branch 'master' into chore/add-specs-for-browser-wallet-sprint

dalebennett1992 committed:
Merge pull request #1696 from vegaprotocol/chore/add-specs-for-browser-wallet-sprint

chore: add ac for wallet browser sprint

mattrussell36 committed:
chore: add acs for market selector

mattrussell36 committed:
Merge pull request #1662 from vegaprotocol/6001-MARK-find_markets

Add UI spec for closed markets

radoslaw-szpiech committed:
chore: added ACs for transaction log

radoslaw-szpiech committed:
chore: removed navigation AC

radoslaw-szpiech committed:
Merge pull request #1701 from vegaprotocol/chore/desktop-wallet-transaction-log

chore: added ACs for transaction log

gordsport committed:
Merge branch 'master' into feat/add-stop-orders-2

gordsport committed:
fix: linting

ValentinTrinque commented:
At the very top, there is mention of:

https://github.com/vegaprotocol/specs/blob/eb06c5e98460f96c733be155af0460870c29c2ef/protocol/0014-ORDT-order_types.md?plain=1#L10

Now the spec is being written, this probably should be removed.

TomMcL commented on protocol/0014-ORDT-order_types.md:
```diff
@@ -47,8 +47,61 @@ Notes on scope of current version of this spec:
+
+  - A network parameter will control the maximum number of stop orders per party (suggested initial value: between 4 and 10).
+
+  - TODO: confirm this makes sense) If the trader's position size gets to zero or changes direction all their stop orders are cancelled.
```
Unsure on this one. The reduce-only-ness of the order should be validated at trade time rather than stop creation time, so it's possible the position touches 0 one way or another between creation and execution. That said, it doesn't seem to *logically* make much sense to have a stop order, close your position/change sides then want it still to trigger. 

Thinking about it though I could see a use for it so I'd probably lean towards not closing them in this case (for example, one might put in both a large buy and sell for if BTC drops below `x` price knowing that only one will execute but you will close out your position whatever happens)

TomMcL committed:
feat: Adding ACs for stop order specs

MuthuVega commented on protocol/0014-ORDT-order_types.md:
```diff
@@ -90,6 +142,45 @@ Network orders are used during [position resolution](./0012-POSR-position_resolu
+
+- A stop order placed either prior to or during an auction will not execute during an auction, nor will it participate in the uncrossing. (<a name="0014-ORDT-032" href="#0014-ORDT-032">0014-ORDT-032</a>)
+- A stop order placed either prior to or during an auction, where the uncrossing price is within the triggering range, will immediately execute following uncrossing. (<a name="0014-ORDT-033" href="#0014-ORDT-033">0014-ORDT-033</a>)
+
```
The AC's for stop loss orders look good to me.

MuthuVega commented:
The AC's for stop loss orders look good to me.

TomMcL committed:
feat: Adding specification for closing stop orders when position and orders are 0

TomMcL commented:
Closing as GitHub caused chaos changing PR branch base, new PR is https://github.com/vegaprotocol/specs/pull/1735
